dbt/adapters/athena/impl.py

```diff
@@ -731,3 +732,97 @@ def get_columns_in_relation(self, relation: AthenaRelation) -> List[AthenaColumn
         return [
             AthenaColumn(column=c["Name"], dtype=c["Type"], table_type=table_type) for c in columns + partition_keys
         ]
+
+    @available.parse_none
+    def valid_snapshot_target(self, relation: BaseRelation) -> None:
+        """Log an error to help developers migrate to the new snapshot logic"""
+        super().valid_snapshot_target(relation)
+        columns = self.get_columns_in_relation(relation)
+        names = {c.name.lower() for c in columns}
+
+        table_columns = [col for col in names if not col.startswith("dbt_") and col != "is_current_record"]
+
+        if "dbt_unique_key" in names:
+            sql = self._generate_snapshot_migration_sql(relation=relation, table_columns=table_columns)
+            msg = (
+                f"{'!'*90}\n"
+                "The snapshot logic of dbt-athena has changed in an incompatible way to be more consistent "
+                "with the dbt-core implementation.\nYou will need to migrate your existing snapshot tables to be "
+                "able to keep using them with the latest dbt-athena version.\nYou can find more information "
+                "in the release notes:\nhttps://github.com/dbt-athena/dbt-athena/releases\n"
+                f"{'!'*90}\n\n"
+                "You can use the example query below as a baseline to perform the migration:\n\n"
+                f"{'-'*90}\n"
+                f"{sql}\n"
+                f"{'-'*90}\n\n"
+            )
+            print(msg)
```

Comment:
Can you remove this or put it as logging debug if needed ?

Reply:
It's a print on purpose. If I log it, then the messages are prefixed with timestamps. That makes it very difficult to copy/paste the query. So it is printed first, and then also logged at the end

Reply:
But if it in the exception, it should be easy to copy / paster it no ?
It is just that I find it weird to have this `print` but if you tell me there is no other way we will keep it like that.

Reply:
> It's a print on purpose. If I log it, then the messages are prefixed with timestamps. That makes it very difficult to copy/paste the query. So it is printed first, and then also logged at the end
that is a fair point, but at the same time the logged query will appear in the stdout when we run dbt models, if some people use json parsing that will fuck up the output.

Let's then please consider to remove it and replace with logger.debug to unify that approach on the all rest of the adapter. Consistency first :)

Reply:
Right. I'm fine with changing to proper logging output. I just noticed that even if I use proper logging, it doesn't add the time information upfront. It only does that for the exception message apparently. So, currently the query is printed twice: once in the final error message, and once in the print statement. How do you suggest to move forward?

- Do we want to keep the query in the final error? We can skip it there, because it's relatively hard to parse anyway. You need to remove the timestamp prefix from each line. With a bit of vim magic that goes quite fast, but I think most users will find this very annoying. Just to give you an example snippet:
  ```
  15:30:37    -- Backup original table
  15:30:37    create table "awsdatacatalog"."blabla"."cc_all_snapshot__dbt_tmp_migration_backup"
  15:30:37      with (
  15:30:37        table_type='hive',
  15:30:37        is_external=true,external_location='s3://some-bucket/tables/blabla/cc_all_snapshot__dbt_tmp_migration_backup/70d57c5a-ea3f-4542-9d13-1b214af36c9f',
  15:30:37        format='parquet'
  15:30:37      )
  15:30:37      as
  15:30:37        select * from "awsdatacatalog"."blabla"."cc_all_snapshot";
  ```
  - Do we want to keep the alternative logging statement? If I set it to debug, and don't show it in the error, then it will be very hard for people to understand what they need to do. If I log on info or warning, then the multiline output has no log prefixes. Not sure how json parsers deal with that tbh. 

I suggest the following: leave the query in the error output and replace the print statement with a debug statement. In the error output I'll then mention that the query is easier available when running in debug. Wdyt?

Reply:
It seems reasonable to me.
But the error is occurring right after the debug right ? 
So we might want to remove the query from the final error.

Reply:
But then it won't be visible without the debug run. That might be annoying for developers that can't directly access the dbt environment right?

Reply:
> I suggest the following: leave the query in the error output and replace the print statement with a debug statement. In the error output I'll then mention that the query is easier available when running in debug. Wdyt?
I think that is fine to use for this `logger.error` the most visible the better, maybe in the logger.error message we can write to switch to debug mode to have a more deeper look at the logs.

Reply:
So it seems that if we log a multiline statement, it will look like this:
```
00:00:00 line 1
line 2
line 3
```

But it seems like dbt captures the exception message if something fails, and reformats the message before printing, so that it looks like this:

```
00:00:00 line 1
00:00:00 line 2
00:00:00 line 3
```

I'm quite fond of this:
```
logger.error(msg)
raise SnapshotMigrationRequired("Look into 1.5 dbt docs for a complete migration procedure")
```